File: Samples for Solid Edge/[C#]-Samples for Solid Edge/C#/SolidEdge.GlobalParameters/StringExtensions.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;

namespace SolidEdge.GlobalParameters
{
    public static class StringExtensions
    {
        public static string CamelCaseToWordString(this string str)
        {
            return Regex.Replace(Regex.Replace(str, @"(\P{Ll})(\P{Ll}\p{Ll})", "$1 $2"), @"(\p{Ll})(\P{Ll})", "$1 $2");
        }
    }
}

File: Samples for Solid Edge/[VB]-Samples for Solid Edge/VB/SolidEdge.GlobalParameters/StringExtensions.vb
Imports System.Text
Imports System.Text.RegularExpressions

Namespace SolidEdge.GlobalParameters
	Public Module StringExtensions
		<System.Runtime.CompilerServices.Extension> _
		Public Function CamelCaseToWordString(ByVal str As String) As String
			Return Regex.Replace(Regex.Replace(str, "(\P{Ll})(\P{Ll}\p{Ll})", "$1 $2"), "(\p{Ll})(\P{Ll})", "$1 $2")
		End Function
	End Module
End Namespace
